### PR TITLE
Feat/cb2 3183 productionise repo

### PR DESCRIPTION
**Pull request by premghinde**
## NOP receives Tech Record Data from new DB

Implementing agency linter and fixes flagged issues
[CB2-3183](https://dvsa.atlassian.net/browse/CB2-3183)

## Checklist

- [ ] Code has been tested manually
- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Comment by premghinde** [comment]
Unit test coverage report
File                | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
--------------------|---------|----------|---------|---------|-------------------
All files           |   96.38 |       80 |   88.23 |   96.25 |                   
 functions          |     100 |       90 |     100 |     100 |                   
  edh-marshaller.ts |     100 |       90 |     100 |     100 | 39                
 models             |     100 |      100 |     100 |     100 |                   
  enums.ts          |     100 |      100 |     100 |     100 |                   
 services           |      90 |    83.33 |   66.66 |      90 |                   
  sqs.ts            |      90 |    83.33 |   66.66 |      90 | 83,93             
 utils              |   96.96 |    57.14 |     100 |   96.66 |                   
  configuration.ts  |     100 |       60 |     100 |     100 | 51                
  utils.ts          |   93.75 |       50 |     100 |    92.3 | 21                

Test Suites: 3 passed, 3 total
Tests:       10 passed, 10 total
Snapshots:   0 total
Time:        8.212 s
Ran all test suites.

**Comment by premghinde** [comment]
Linter report 

/Users/prem/CVS/cvs-tsk-edh-marshaller/src/utils/configuration.ts
  42:22  warning  Unsafe Regular Expression  security/detect-unsafe-regex

✖ 1 problem (0 errors, 1 warning)

**Review by danielchildsbjss — DISMISSED** [review]

**Review by Ian-Hodges — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -47,45 +48,45 @@
...
+    "@typescript-eslint/eslint-plugin": "^5.23.0",
+    "@typescript-eslint/parser": "^5.23.0",
     "aws-lambda-mock-context": "^3.2.1",
-    "eslint": "^6.8.0",
-    "husky": "^3.0.9",
-    "jest": "^24.9.0",
+    "eslint": "^8.15.0",
+    "eslint-config-airbnb-typescript": "^17.0.0",
+    "eslint-import-resolver-typescript": "^2.7.1",
+    "eslint-plugin-jest": "^26.1.5",
```
Comment (aaron-downing) [inline package.json:71]: I've just pulled the branch down and got this after running the lint command.

image

Then after adding the package I got all of these :( 

image

Comment (aaron-downing) [reply package.json:71]: The tests also fail due to referencing issues

Comment (aaron-downing) [reply package.json:71]: Fixing the references resolves the linting issues 👍 but the eslint-plugin-import package also needs adding

**Review by aaron-downing — DISMISSED** [review]

**Review by kpearmanbjss — DISMISSED** [review]

File: package.json [hunk 2]
```diff
@@ -13,14 +13,15 @@
...
+    "test:unit": "BRANCH=local SLS_DEBUG=* AWS_XRAY_CONTEXT_MISSING=LOG_ERROR jest --testMatch=\"**/*.test.ts\" --runInBand",
     "test": "npm run test:unit -- --coverage",
     "test-i": "echo 'nothing to do'",
-    "lint": "eslint . --fix",
+    "lint": "eslint .",
+    "lint:fix": "eslint . --fix",
     "format": "prettier --write .",
     "sonar-scanner": "sonar-scanner",
     "prepush": "npm run test && npm run build && npm run test-i",
-    "security-checks": "git secrets --scan && git log -p | scanrepo",
```
Comment (gjulien-bjss) [inline package.json:23]: If scanrepo is not used anymore should it be removed as project dependencies/prerequisites in the readme?

**Review by gjulien-bjss — APPROVED** [review]

**Review by aaron-downing — APPROVED** [review]

**Pull request merged**